packages/dds/tree/src/simple-tree/types.ts

```diff
@@ -72,6 +135,39 @@ export interface TreeChangeEvents {
 	 * - For a map node, when an entry is added, updated, or removed.
 	 *
 	 * @remarks
+	 * Prefer using events on the specific node kind (object nodes, map nodes, array nodes) instead of this one
+	 * for more specific listener signatures according to what a given node kind actually supports.
+	 *
+	 * This event is not emitted when:
+	 *
+	 * - Properties of a child node change. Notably, updates to an array node or a map node (like adding or removing
+	 * elements/entries) will emit this event on the array/map node itself, but not on the node that contains the
+	 * array/map node as one of its properties.
+	 *
+	 * - The node is moved to a different location in the tree or removed from the tree.
+	 * In this case the event is emitted on the _parent_ node, not the node itself.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
+	 * the client that made the original edit.
+	 *
+	 * When it is emitted, the tree is guaranteed to be in-schema.
+	 *
+	 * @privateRemarks
+	 * This event occurs whenever the apparent contents of the node instance change, regardless of what caused the change.
+	 * For example, it will fire when the local client reassigns a child, when part of a remote edit is applied to the
+	 * node, or when the node has to be updated due to resolution of a merge conflict
+	 * (for example a previously applied local change might be undone, then reapplied differently or not at all).
+	 */
+	nodeChanged(): void;
+}
+
+export interface ObjectNodeChangeEvents {
+	/**
```

Comment:
Don't know if there's a better way to reuse all this documentation that needs little tweaks in each case.